Add CONTRIBUTING.md Guidelines

kwindeezah opened the pull request:
## Description  
Added a CONTRIBUTING.md file in the root directory

## Related Issue  
**Fixes #84**
**Related to #84**

## Type of Change  
- [x] Documentation or Content update

## Checklist  
- [x] I have updated documentation (if applicable)  
- [x] My changes follow the project's code style and contribution guidelines

Commit:
added dropdown menu do navigate between ressource page

Commit:
updating header file

kwindeezah committed:
docs: add CONTRIBUTING.md guidelines

Commit:
adding routes and adding new design to the hoover

Commit:
updated route

MutiatBash committed:
Merge branch 'main' into docs/contributing-guide

Commit:
updtating links and text

Mayaleeeee commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,141 @@
+
+1. Go to the [Open Nest GitHub page](https://github.com/open-nest-africa/open-nest-africa).
+2. Click the **Fork** button in the top right corner.
+3. This will create a copy of the repository under your GitHub account.
```
Thanks for working on this, @kwindeezah 

Please include a screenshot of the locations you are referencing to make it easier to follow. This will aid in comprehension and serve as a helpful accessibility guide.

kwindeezah replied:
Noted.

Mayaleeeee commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,141 @@
+   - Ask your question in the Q&A discussions category
+   - Reach out via [project community channels](https://opennestafrica.slack.com)
+
+**Thank you for contributing to Open Nest! 🚀🫶🏼**
```
Also, please change the name to `OpenNest-Africa`, not Open Nest.

kwindeezah replied:
Noted, thanks for the feedback.

kwindeezah committed:
docs: update CONTRIBUTING.md guidelines

kwindeezah committed:
Merge remote-tracking branch 'upstream/main' into docs/contributing-guide

kwindeezah committed:
docs: update CONTRIBUTING.md guidlines

kwindeezah committed:
docs: update CONTRIBUTING.md guidlines

kwindeezah committed:
docs: update CONTRIBUTING.md guidelines

Commit:
updating ressources file

Commit:
removed extra t on the developers,jsx file

MutiatBash committed:
Merge branch 'main' into docs/contributing-guide

MutiatBash commented on CONTRIBUTING.md:
```diff
@@ -145,6 +145,5 @@ If you have questions or want to discuss ideas:
    - Reach out via [project community channels](https://opennestafrica.slack.com)
-   ![Discussions](public/images/discussions.png)
 
 **Thank you for contributing to Open Nest Africa! 🚀🫶🏼**
```
Hello @kwindeezah ......It seems you've not corrected the open nest...it should be **OpenNest-Africa**...Thank you

MutiatBash replied:
@kwindeezah Hi, if you are still interested in contributing, please respond.
If we don't get feedback from you in 3 days..we'll be closing your PR..Thank you

kwindeezah replied:
Ok, I'll make the correction.

Commit:
updtating header

Thirukumaran-T committed:
REFACTOR: added styles to make footer responsiveness

MutiatBash committed:
Revert " added dropdown menu do navigate between ressource page"

Commit:
added dropdown menu to ressource page

Thirukumaran-T committed:
CHORE: added styles to make the footer responsive

olisaagbafor committed:
feat: implement password reset functionality

- Add password reset endpoints and services
- Integrate email service for reset links
- Add secure token generation and validation
- Adjusted GitHub auth and user profile endpoints

olisaagbafor committed:
Restructured Backend & Implemented Password Request/Reset pages

Mayaleeeee committed:
Create greetings.yml

Mayaleeeee committed:
Update README.md

Mayaleeeee committed:
Update README.md

Mayaleeeee committed:
Update README.md

Mayaleeeee committed:
Update README.md

Mayaleeeee committed:
Update greetings.yml

Iwueseiter committed:
feat: implemented responsiveness on the landing page

GifftyCode committed:
Feat: Implement Google Auth (Sign Up & Sign In) with Firebase

Commit:
updtating header file

olisaagbafor committed:
Fix: Reset Password Flow Modifications
- Update TextField component to include onChange property
- Remove unused Console Log and Comments
- Modified User Model to allow no password during password request

GifftyCode committed:
fix: handle firebase errors with user-friendly messages

olisaagbafor committed:
feat: AuthLayout enhancement
- Added showOpenNestText prop to the Authlayout
- Properly implemented the show/hide password icon for password fields
- Removed the sub-headings in the Reset Password pages

olisaagbafor committed:
Removed comments in server.js and auth service file

olisaagbafor committed:
Added env for API_URL

Iwueseiter committed:
feat: implemented the landing page

MutiatBash committed:
Removed comment from authservice

MutiatBash committed:
password reset test and update

MutiatBash committed:
fix: remove unnecessary div from header component and fix build error

MutiatBash committed:
fix: correct class name for desktop navigation in header component

Iwueseiter committed:
feat: resolved merge conflicts

Iwueseiter committed:
feat: resolved merge conflicts

Iwueseiter committed:
fixed footer

GifftyCode committed:
fix: package.lock merge conflict

GifftyCode committed:
package-lock.json merge conflicts

GifftyCode committed:
packages installed

Marvellous-Udoye committed:
feat: implement hover effects to buttons and links for better interactivity

MutiatBash committed:
Revert "Feat: Implement Google Auth (Sign Up & Sign In) with Firebase"

Marvellous-Udoye committed:
feat: update loading condition for repository list

Marvellous-Udoye committed:
feat: update hover effect for links and auth buttons

oliverTwist2 committed:
Added nodemon as a dev dependency, changed the dev start script to use npm run dev in backend. Removed process.exiit() in database connection so server doesn't break if mongo connection fails

oliverTwist2 committed:
Corrected the name of .env.example in the root directory. Added .env.example file for backend directory and populated it with all the env variables used in the backend.

oliverTwist2 committed:
Added unit tests for all the controller and service logic with jest in test folder. All tests passed.

Marvellous-Udoye committed:
style: add hover effects to nav links

oliverTwist2 committed:
edited gitignore to allow .env.example to upload to github

jumalaw98 commented:
@kwindeezah are you able to also add, someone as a designer, how they can access Figma file, to suggest or direct contribute on designs

jumalaw98 commented:
@Mayaleeeee is it also possible we can add task duration for every first issue, to avoid essence of one assigned task and goes missing, with no communication of progress of the tasks?

kwindeezah commented:
> are you able to also add, someone as a designer, how they can access Figma file, to suggest or direct contribute on designs

@jumalaw98 Yes, I can include that.
Please share the Figma file or let me know where I can access it, and I’ll add the relevant details.
Thank you!

kwindeezah commented:
> > > are you able to also add, someone as a designer, how they can access Figma file, to suggest or direct contribute on designs
> > 
> > 
> > @jumalaw98 Yes, I can include that. Please share the Figma file or let me know where I can access it, and I’ll add the relevant details. Thank you!
> 
> here it is: https://www.figma.com/design/ajDY4kQpZkCpnjgQdPmWZM/OpenNest-Africa--Copy-?node-id=2005-1600&t=HQUhwiT3RKGjyCas-0

Thanks

kwindeezah committed:
Docs: Update with designer guide

kwindeezah committed:
Merge remote-tracking branch 'upstream/dev' into docs/contributing-guide

kwindeezah commented:
@jumalaw98 I have included the details for designers.
Kindly review.

jumalaw98 reviewed (dismissed):
This CONTRIBUTING.md is ready to be merged into the main branch. 
Great work! 👏👏👏

Mayaleeeee commented:
>  This CONTRIBUTING.md is ready to be merged into the main branch. 
> Great work! 👏👏👏

@jumalaw98 

Please don't approve any changes until a Maintainer has given the instructions to do so.

MutiatBash committed:
Merge branch 'dev' into docs/contributing-guide

MutiatBash reviewed (commented):
Hello @kwindeezah ......It seems you've not corrected the open nest...it should be OpenNest-Africa...Thank you
